packages/calcite-components/src/components/date-picker-month-header/date-picker-month-header.tsx

```diff
@@ -323,16 +408,127 @@ export class DatePickerMonthHeader {
     commit?: boolean;
     offset?: number;
   }): void {
-    const { yearInput, activeDate } = this;
+    const { yearInputEl, activeDate } = this;
     const inRangeDate = this.getInRangeDate({ localizedYear, offset });
 
     // if you've supplied a year and it's in range, update active date
     if (inRangeDate) {
-      this.calciteInternalDatePickerSelect.emit(inRangeDate);
+      this.calciteInternalDatePickerMonthHeaderSelect.emit(inRangeDate);
     }
 
     if (commit) {
-      yearInput.value = this.formatCalendarYear((inRangeDate || activeDate).getFullYear());
+      yearInputEl.value = this.formatCalendarYear((inRangeDate || activeDate).getFullYear());
+    }
+  }
+
+  private async setSelectMenuIconOffset(select: HTMLCalciteSelectElement): Promise<void> {
+    const iconEl = select.shadowRoot.querySelector("calcite-icon");
+    await componentOnReady(iconEl);
```

Comment:
Child components should be ready at this point (see https://stenciljs.com/docs/component-lifecycle#lifecycle-hierarchy). If not, there might be a watcher being explicitly invoked.

Reply:
Yup, the `value` Watcher in `select` is invoked .